cmd/gcp-controller-manager/app/gke_approver.go

```diff
@@ -673,6 +673,43 @@ func ensureNodeMatchesMetadataOrDelete(opts GCPConfig, csr *capi.CertificateSign
 }
 
 func shouldDeleteNode(opts GCPConfig, node *v1.Node) (bool, error) {
-	// TODO: return true if the Node API object does not match metadata and should be deleted
+	inst, err := getInstanceByName(opts, node.Name)
+	if err != nil {
+		glog.Errorf("Error retrieving instance %q: %v. Falling back to not deleting the node: %v", node.Name, err)
+		return false, nil
```

Comment:
return false, err
to make approver retry instead of approving

Reply:
How many times would approver retry before giving up? Was trying to avoid hanging the node forever if GCE is acting weird.

Reply:
I don't think it gives up.
Our Node startup SLO should be lower than GCE API SLO. It should be fine to hang on GCE API issues.

With current approach you're also risking registering a Node with the podCIDR issue which makes this fix unreliable.

Reply:
Makes sense, I will update the commit.